### PR TITLE
Update Feature flags documentation

### PR DESCRIPTION
**Pull request by ApexModder**
This pr fixes couple things that were brought to may attention as people made use of feature flags.

- Fixed json syntax error in the `#feature-packs` section
- Added a note about the `pack.mcmeta` file not being the same as the mods root `pack.mcmeta` file.
- Updates the default file location in `gettingstarted/modfiles.md` to match `advanced/featureflags.md`

------------------
Preview URL: https://pr-199.neoforged-docs-previews.pages.dev

### CODE REVIEW

File: docs/advanced/featureflags.md [hunk 1]
```diff
@@ -157,15 +157,20 @@ _See also: [Resource Packs](../resources/index.md#assets), [Data Packs](../resou
 
 Feature packs are a type of pack that not only loads resources and/or data, but also has the ability to toggle on a given set of feature flags. These flags are defined in the `pack.mcmeta` JSON file at the root of this pack, which follows the below format:
 
+:::note
+This file differs from that of the one in your mods `resources/` directory. This file defines a brand new feature pack and as thus must be in its own folder.
```
Comment (IchHabeHunger54) [inline docs/advanced/featureflags.md:161]:
```suggestion
This file differs from that of the one in your mod's `resources/` directory. This file defines a brand new feature pack and as thus must be in its own folder.
```

**Review by IchHabeHunger54 — CHANGES_REQUESTED** [review]
One typo I found, good otherwise.

File: docs/advanced/featureflags.md [hunk 2]
```diff
@@ -157,15 +157,20 @@ _See also: [Resource Packs](../resources/index.md#assets), [Data Packs](../resou
 
 Feature packs are a type of pack that not only loads resources and/or data, but also has the ability to toggle on a given set of feature flags. These flags are defined in the `pack.mcmeta` JSON file at the root of this pack, which follows the below format:
 
+:::note
+This file differs from that of the one in your mod's `resources/` directory. This file defines a brand new feature pack and as thus must be in its own folder.
```
Comment (XFactHD) [inline docs/advanced/featureflags.md:161]:
```suggestion
This file differs from the one in your mod's `resources/` directory. This file defines a brand new feature pack and as thus must be in its own folder.
```

File: docs/advanced/featureflags.md [hunk 3]
```diff
@@ -157,15 +157,20 @@ _See also: [Resource Packs](../resources/index.md#assets), [Data Packs](../resou
 
 Feature packs are a type of pack that not only loads resources and/or data, but also has the ability to toggle on a given set of feature flags. These flags are defined in the `pack.mcmeta` JSON file at the root of this pack, which follows the below format:
 
+:::note
+This file differs from the one in your mod's `resources/` directory. This file defines a brand new feature pack and as thus must be in its own folder.
```
Comment (XFactHD) [inline docs/advanced/featureflags.md:161]:
```suggestion
This file differs from the one in your mod's `resources/` directory. This file defines a brand new feature pack and 
thus must be in its own folder.
```

Comment (ApexModder) [reply docs/advanced/featureflags.md:161]: Why add the new line here? this goes against the rest of the file and overall format from what ive seen, where everything is on 1 line.
Also feels weird to me to add new line midway through a sentence.

Comment (XFactHD) [reply docs/advanced/featureflags.md:161]: No idea what GitHub did there, the intention was just to delete a misplaced word. Fixed the suggestion.

**Review by XFactHD — APPROVED** [review]

**Review by IchHabeHunger54 — APPROVED** [review]

**Pull request merged**